src/Behavioral.Automation/Services/AutomationIDProvider.cs

```diff
@@ -10,22 +13,41 @@ namespace Behavioral.Automation.Services
     public sealed class AutomationIdProvider : IAutomationIdProvider
     {
         private readonly IScopeContextRuntime _scopeContextRuntime;
+        private readonly IDriverService _driverService;
+        private readonly IScopeContextManager _scopeContextManager;
 
-        public AutomationIdProvider([NotNull] IScopeContextRuntime scopeContextRuntime)
+        public AutomationIdProvider(
+            [NotNull] IScopeContextRuntime scopeContextRuntime,
+            [NotNull] IDriverService driverService,
+            [NotNull] IScopeContextManager scopeContextManager)
         {
+            _scopeContextManager = scopeContextManager;
+            _driverService = driverService;
             _scopeContextRuntime = scopeContextRuntime;
         }
 
 
-        /// <summary>
-        /// Get control description by element caption
-        /// </summary>
-        /// <param name="caption">Caption of the tested element</param>
+        /// <summary>
+        /// Get control description by element caption
+        /// </summary>
+        /// <param name="caption">Caption of the tested element</param>
         /// <returns>Element control description containing its name, type, id and subpath</returns>
         public ControlDescription Get(string caption)
         {
             ParseCaption(caption, out var name, out var type);
-            return _scopeContextRuntime.FindControlDescription(type, name);
+
+            var controlDescription = _scopeContextRuntime.FindControlDescription(type, name);
+
+            if (controlDescription != null) 
+                return controlDescription;
+
+            Thread.Sleep(500);
```

Comment:
What do you think, one attempt will be enough?

Reply:
I performed several test runs (up to 800 tests for each) and context related exceptions had never been raised, so I think 500 ms is enough

Reply:
Ok. In case of a similar issue, we can add several attempts with a 500 ms delay.